### PR TITLE
⚡️ Rewrite FixedPointMathLib sqrt() in Yul

### PR DESCRIPTION
**Pull request by ZeframLou**

### CODE REVIEW

File: src/utils/FixedPointMathLib.sol [hunk 1]
```diff
@@ -120,56 +120,88 @@ library FixedPointMathLib {
     //////////////////////////////////////////////////////////////*/
 
     function sqrt(uint256 x) internal pure returns (uint256 result) {
-        if (x == 0) return 0;
+        assembly {
+            switch x
```
Comment (transmissions11) [inline src/utils/FixedPointMathLib.sol:124]: i wonder if a not zero check would be cheaper than an switch?

**Comment by transmissions11** [comment]
reminder to self: gotta port over the comments from `audit-fixes` too

File: src/utils/FixedPointMathLib.sol (hunk 1 above)
Comment (ZeframLou) [reply src/utils/FixedPointMathLib.sol:124]: true true, one sec

Comment (transmissions11) [reply src/utils/FixedPointMathLib.sol]: wow wtf that saved a lot more than i was expecting, i wanna take a look in the hevm debugger in the morning

**Comment by transmissions11** [comment]
also could we just do if x or is the double iszero() necessary 👁️

**Comment by transmissions11** [comment]
> also could we just do if x or is the double iszero() necessary 👁️

checked, optimizer does it for us.

**Pull request merged**